src/screens/Manager/Manager.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useCallback, useEffect } from "react";
+import { NavigationActions } from "react-navigation";
+import type { Action, State } from "@ledgerhq/live-common/lib/apps";
+
+import { ManagerContext } from "./shared";
+import AppsScreen from "./AppsScreen";
+import GenericErrorBottomModal from "../../components/GenericErrorBottomModal";
+import QuitManagerModal from "./Modals/QuitManagerModal";
+import StorageWarningModal from "./Modals/StorageWarningModal";
+import AppDependenciesModal from "./Modals/AppDependenciesModal";
+import UninstallDependenciesModal from "./Modals/UninstallDependenciesModal";
+
+type Props = {
+  screenProps: {
+    state: State,
+    dispatch: Action => void,
+  },
+  navigation: *,
+};
+
+export default ({ screenProps: { state, dispatch }, navigation }: Props) => {
+  const { apps, currentError, installQueue, uninstallQueue } = state;
+  const [quitManagerAction, setQuitManagerAction] = useState(false);
+
+  useEffect(() => {
+    const blockNavigation = installQueue.length + uninstallQueue.length > 0;
+    const n = navigation.dangerouslyGetParent();
+    if (n) {
+      n.setParams({ blockNavigation });
+      let navListener;
+      if (blockNavigation) {
+        navListener = navigation.addListener("action", e => {
+          if (e.action && e.action.type !== NavigationActions.SET_PARAMS) {
+            setQuitManagerAction(e.action);
+          }
+        });
+      } else if (navListener) {
+        navListener.remove();
```

Comment:
I don't understand how this can work. In this `else` branch, `navListener` is always falsy.

I think what you need to do instead is to `return () => navListener.remove()` in the first if block?